balancer/balanceraggregator/balanceraggregator.go

```diff
@@ -0,0 +1,265 @@
+/*
+ *
+ * Copyright 2024 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package balanceraggregator implements a BalancerAggregator helper.
+package balanceraggregator
```

Comment:
Bikeshedding is needed, I think.

How about `endpointmanager'?  `endpointshard[ing]`?

Also let's mark this as experimental for now.

Reply:
Added:
//
// # Experimental
//
// Notice: This package is EXPERIMENTAL and may be changed or removed in a
// later release.

The more terse comment I saw was "// All APIs in this package are experimental." Let me know if you would prefer that instead.

Reply:
The terse comment is older, I think.  The bigger one formats more nicely in godoc.

Reply:
Alright I'll leave it as is then.

Reply:
"blah blah blah the EndpointSharding balancer blah blah" maybe?  Seems better than "EndpointShard" and "EndpointSharder"  I'm also okay with EndpointManager but I like what "sharding" implies better.  Also open to other ideas.

Reply:
I think EndpointsSharding or EndpointListSharding. EndpointSharding singular implies that you're sharding a single endpoint?

Reply:
Or even EndpointListBalancer, as C++ does. I'll switch it to EndpointSharding for the time being. I also kind of see that the Endpoint can be referring to what is the thing that is the output of the sharding (the atom). We can discuss this more, I'm fine with whatever.

Reply:
Yeah, it's "sharded _by endpoint_".

I like this name.  I'm also okay with `balancer/endpointlist.NewEndpointListBalancer`.  `EndpointSharded` could also work.